src/opm/parser/eclipse/EclipseState/Schedule/Group/Group2.cpp

```diff
@@ -22,9 +22,239 @@
 #include <opm/parser/eclipse/EclipseState/Schedule/Group/Group2.hpp>
 
 namespace Opm {
+Group2::Group2(const std::string& name, std::size_t insert_index_arg, std::size_t init_step_arg) :
+    m_name(name),
+    m_insert_index(insert_index_arg),
+    init_step(init_step_arg),
+    group_type(GroupType::NONE),
+    gefac(1),
+    transfer_gefac(true),
+    vfp_table(0)
+{
+    // All groups are initially created as children of the "FIELD" group.
+    if (name != "FIELD")
+        this->parent_group = "FIELD";
+}
+
+std::size_t Group2::insert_index() const {
+    return this->m_insert_index;
+}
+
+bool Group2::defined(size_t timeStep) const {
+    if (timeStep < this->init_step)
+        return false;
+    else
+        return true;
```

Comment:
I'll never understand why you think that is more readable than
```C++
return timeStep >= this->init_step;
```

Reply:
I'll update - this one I really do not care all that much about ;-)